core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -256,9 +257,30 @@ public UfsStatus[] listStatus(String path, ListOptions options) throws IOExcepti
     return statuses;
   }
 
+  /**
+   * @param fileInfo the FileInfo of this file. Cached pages are identified by PageId
+   * @return true at this moment
+   */
+  @Override
+  public boolean invalidateCachedFile(FileInfo fileInfo) {
+    long pages = (fileInfo.getLength() + mPageSize - 1) / mPageSize;
+    FileId file = FileId.of(new AlluxioURI(fileInfo.getUfsPath()).hash());
+
+    // @TODO(huanghua78) Only invalidate cached pages. Here, we don't check if a page is
+    // cached or not. If a page is not cached at all, the mCacheManager.delete() will log
+    // error messages to complain the non-existance of the page. As an optimization we need
+    // to only invalidate cached pages.
+    for (long i = 0; i < pages; i++) {
+      PageId page = new PageId(file.toString(), i);
+      mCacheManager.delete(page);
```

Comment:
Feeling like it can be helpful to have existing cache manager for page level operations, and another layer on top for file level operations, the page calculation is duplicated in many places, but it's something out of scope of this PR

Reply:
Discussed this with @dbw9580 .
Because the CacheManager only has PageId related interface. So I have to do the calculation here.
But as you said, if the CacheManager provides file level operation, I can call something like:
mCacheManager.delete(filename);